histogram.go

```diff
@@ -0,0 +1,42 @@
+package imaging
+
+import (
+	"image"
+)
+
+// Histogram returns a normalized histogram of an image.
+//
+// Resulting histogram is represented as an array of 256 floats, where
+// histogram[i] is a probability of a pixel being of a particular luminance i.
+func Histogram(img image.Image) [256]float32 {
+	src := toNRGBA(img)
+	width := src.Bounds().Max.X
+	height := src.Bounds().Max.Y
+
+	var histogram [256]float32
+	var total float32
+
+	parallel(height, func(partStart, partEnd int) {
+		for y := partStart; y < partEnd; y++ {
+			for x := 0; x < width; x++ {
+				i := y*src.Stride + x*4
+
+				r := src.Pix[i+0]
+				g := src.Pix[i+1]
+				b := src.Pix[i+2]
+
+				var y float32 = 0.2126*float32(r) + 0.7152*float32(g) + 0.0722*float32(b)
+				y = y
+
+				histogram[int(y)]++
```

Comment:
Maybe it's better to use rounding to int instead of truncating e.g. `int(y+0.5)`? So that luminance of 254.9 will be counted as 255 and not as 254. What do you think?